source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py

```diff
@@ -238,7 +238,16 @@ def build_ridesharing_journeys(self, from_pt_obj, to_pt_obj, period_extremity):
             start_teleport_section.origin.CopyFrom(from_pt_obj)
             start_teleport_section.destination.CopyFrom(pb_rsj_pickup)
             start_teleport_section.length = int(crowfly_distance_between(from_coord, pickup_coord))
-            start_teleport_section.duration = 0
+
+            # We take departure to pickup duration
+            start_teleport_section.duration = (
+                int(rsj.origin_pickup_duration)
+                if getattr(rsj, 'origin_pickup_duration', None) is not None
+                else 0
```

Comment:
Maybe simpler like this ? :
```suggestion
                int(getattr(rsj, 'origin_pickup_duration', 0))
```

Reply:
:+1: